dotCMS/src/main/java/com/dotcms/content/elasticsearch/ESQueryCache.java

```diff
@@ -0,0 +1,103 @@
+package com.dotcms.content.elasticsearch;
+
+import java.util.Arrays;
+import java.util.Optional;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.search.SearchHits;
+import com.dotcms.enterprise.license.LicenseManager;
+import com.dotmarketing.business.Cachable;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotCacheAdministrator;
+import com.dotmarketing.util.Config;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.hash.Hashing;
+
+public class ESQueryCache implements Cachable {
+
+
+    private final DotCacheAdministrator cache;
+
+    private final boolean shouldUseCache;
+
+    @VisibleForTesting
+    public ESQueryCache(DotCacheAdministrator cache, boolean shouldUseCache) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'shouldUseCache' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5288908)